dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/test-applications/thirdpartylib/src/com/ibm/ws/jaxrs2x/fat/thirdparty/multipart/UploadServiceImpl.java

```diff
@@ -70,6 +73,37 @@ public Response uploadFile(List<IAttachment> attachments, @Context HttpServletRe
         return Response.ok("file uploaded").build();
     }
 
+    @POST
+    @Path("/uploadFile2")
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public Response uploadFile2(List<IAttachment> attachments, @Context HttpServletRequest request) {
+        for (IAttachment attachment : attachments) {
+            DataHandler handler = attachment.getDataHandler();
+            try {
+                InputStream stream = handler.getInputStream();
+                MultivaluedMap<String, String> map = attachment.getHeaders();
+                System.out.println("fileName2 Here " + getFileName(map));
+                OutputStream out = new FileOutputStream(new File("./" + getFileName(map)));
+                StringBuilder stringBuilder = new StringBuilder();
+                Reader in = new InputStreamReader(stream, StandardCharsets.UTF_8);
+                char[] chars = new char[1024];
+                int charsRead;
+                while((charsRead = in.read(chars, 0, chars.length)) > 0) {
+                    stringBuilder.append(chars, 0, charsRead);
+                    out.write(new String(chars).getBytes("UTF-8"), 0, charsRead);
+                }
+                System.out.println("uploadFile2 stringBuilder.toString(): " + stringBuilder.toString());
+                stream.close();
+                out.flush();
+                out.close();
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+
+        return Response.ok("file uploaded").build();
```

Comment:
this will always return ok("file uploaded") even if there was an exception thrown.

Maybe put this return statement as the last line of the try block, and then add the following line inside the catch block (after printing the stack trace): 
`return Response.serverError().entity("caught exception: " + e);`